### PR TITLE
Print attestation token when device not claimed

### PR DESCRIPTION
**Pull request by plskeggs**
Help users with claiming and onboarding their device by displaying the attestation token for them when failure to access to the provisioning service indicates the device is unclaimed. Control this with a Kconfig.

Add overlay to the multi service sample for configuring nrf_provisioning for CoAP. Improve logging to be more helpful.

### CODE REVIEW

File: subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c
Comment (glarsennordic) [inline subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c:582]: Maybe print this always, and print it before the `CONFIG_NRF_PROVISIONING_PRINT_ATTESTATION_TOKEN`

That way if `modem_attest_token_get` fails, we still print something cogent for the user

File: subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c
Comment (glarsennordic) [inline subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c:575]: Should this be printk, or LOG_INFO/LOG_WRN?

File: samples/cellular/nrf_cloud_multi_service/overlay-coap_nrf_provisioning.conf
Comment (glarsennordic) [inline samples/cellular/nrf_cloud_multi_service/overlay-coap_nrf_provisioning.conf:1]: Should the HTTP version of the overlay be renamed to `http_nrf_provisioning`?

I also think docs should probably be updated to mention both overlays

File: samples/cellular/nrf_cloud_multi_service/src/cloud_connection.c
Comment (glarsennordic) [inline samples/cellular/nrf_cloud_multi_service/src/cloud_connection.c:550]: Good thinking

File: subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c
Comment (plskeggs) [reply subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c:575]: I specifically chose printk to make the log of attestation token more obvious and readable. So much of what gets logged is not actionable by the user -- I want this information to stand out, and to not be suppressed by log level.

File: samples/cellular/nrf_cloud_multi_service/overlay-coap_nrf_provisioning.conf
Comment (plskeggs) [reply samples/cellular/nrf_cloud_multi_service/overlay-coap_nrf_provisioning.conf:1]: I haven't done the docs yet, but I agree.

File: subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c
Comment (glarsennordic) [reply subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c:575]: I'd argue this is a good use-case for LOG_ERR or LOG_WRN, then, since those will not usually be suppressed -- That way it stays congruent with the logging subsystem

Comment (plskeggs) [reply subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c:575]: Nope.

File: subsys/net/lib/nrf_provisioning/Kconfig
Comment (jayteemo) [inline subsys/net/lib/nrf_provisioning/Kconfig:78]:
```suggestion
	default y if !NRF_PROVISIONING_SHELL
	depends on NRF_PROVISIONING_LOG_LEVEL_INF
```

prefer to use LOG_INF instead of printk, but i'll leave that up to the code owners.

**Review by jayteemo — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

File: samples/cellular/nrf_cloud_multi_service/overlay-coap_nrf_provisioning.conf
Comment (jayteemo) [inline samples/cellular/nrf_cloud_multi_service/overlay-coap_nrf_provisioning.conf:22]:
```suggestion
# Use a sec tag different than the HTTP overlay since CoAP uses a different CA certificate
CONFIG_NRF_PROVISIONING_ROOT_CA_SEC_TAG=51
```

Comment (plskeggs) [reply samples/cellular/nrf_cloud_multi_service/overlay-coap_nrf_provisioning.conf:22]: I used 42 because the nrf_provisioning sample uses 42 for CoAP.  I was following their lead.

Comment (jayteemo) [reply samples/cellular/nrf_cloud_multi_service/overlay-coap_nrf_provisioning.conf:22]: i suggested 51 since other NCS samples also use 42.

File: samples/cellular/nrf_cloud_multi_service/overlay-coap_nrf_provisioning.conf
Comment (jayteemo) [inline samples/cellular/nrf_cloud_multi_service/overlay-coap_nrf_provisioning.conf:11]: do we want this here?
yes, it is needed for auto-onboarding.
but it differs from HTTP provisioning and default MSS builds.

File: samples/cellular/nrf_cloud_multi_service/overlay-coap_nrf_provisioning.conf
Comment (plskeggs) [reply samples/cellular/nrf_cloud_multi_service/overlay-coap_nrf_provisioning.conf:22]: I know your reason. I am aware of the other use of 42. I stand by my answer.

File: samples/cellular/nrf_cloud_multi_service/overlay-coap_nrf_provisioning.conf
Comment (plskeggs) [reply samples/cellular/nrf_cloud_multi_service/overlay-coap_nrf_provisioning.conf:11]: Yes, I'm surprised it isn't the selection for HTTP.

Comment (jayteemo) [reply samples/cellular/nrf_cloud_multi_service/overlay-coap_nrf_provisioning.conf:11]: it should be removed here.
otherwise docs need updated.

File: subsys/net/lib/nrf_provisioning/Kconfig
Comment (peknis) [inline subsys/net/lib/nrf_provisioning/Kconfig:82]:
```suggestion
	  nRF Cloud. Display the attestation token
	  needed for claiming the device.
```

**Review by juhaylinen — APPROVED** [review]

**Review by peknis — APPROVED** [review]
Approved with a minor suggestion to Kconfig help text.

File: samples/cellular/nrf_cloud_multi_service/overlay-coap_nrf_provisioning.conf
Comment (plskeggs) [reply samples/cellular/nrf_cloud_multi_service/overlay-coap_nrf_provisioning.conf:11]: It will be very soon, but I'll do another PR for that that also updates the REST device message sample.

**Review by glarsennordic — APPROVED** [review]

**Review by tony-le-24 — APPROVED** [review]
All combinations of DIaP have been tested and passed. LGTM!

**Pull request merged**
